app/models/booking_response.rb

```diff
@@ -40,7 +43,34 @@ def no_allowance?
     selection == 'no_allowance'
   end
 
+  def visitor_not_on_list?
+    selection == 'visitor_not_on_list'
+  end
+
+  def visitor_banned?
+    selection == 'visitor_banned'
+  end
+
   def slot_index
     SLOTS.index(selection)
   end
+
+  def unlisted_visitors
+    visitors.select { |v| unlisted_visitor_ids.include?(v.id) }
+  end
+
+  def banned_visitors
+    visitors.select { |v| banned_visitor_ids.include?(v.id) }
+  end
+
+private
+
+  def validate_checked_visitors
+    if selection == 'visitor_not_on_list' && unlisted_visitor_ids.empty?
+      errors.add :selection, :no_unlisted_visitors_selected
```

Comment:
Could these errors be more declarative?  `:no_unlisted_visitors_have_been_selected` or something similar...